### DISCUSSION TITLE
FPS counter hides during gameplay

### DISCUSSION THREAD

**Discussion by SevenChords** [ppy/osu#19859] [Q&A] [answered]
The FPS counter that you can enable in the video settings is only shown in menus and during song breaks for me, which kind of defeats it's purpose since during gameplay is the one part where high/constant fps matters the most, and if the FPS counter hides during gameplay while you suspect fps drops it's kind of impossible to tell if fps are actually dropping with the fps counter hidden. The hud overlay is set to always show in the settings, i would've assumed that includes the fps counter, but apparently not. Game version 2022.816.0-lazer, platform is macos.

Is there some setting i'm missing to not hide the fps counter during gameplay? If so please tell me, cause the way it is right now the fps counter feels completely pointless since it only shows during menus and in song breaks where nothing is happening on screen.

**Comment by Doksha** [comment, marked as answer]
I believe it shows up when there are issues with frames (ie. dropping frames, or lower framerate) not entirely sure, but yeah, it is quite annoying having it constantly showing up then going away in the middle of a map. Probably need more info from devs on what exactly causes it to show.

**Reply by SevenChords** [reply]
i'm at a point rn where i am not sure if the stuttering i'm experiencing is from my screen or from the game, that's the main reason why i would like to have an option to have it constantly displayed

**Reply by peppy** [reply]
If there is a stutter, the FPS counter will show up (even if it's just one frame).

You can confirm this by showing the full frame graph (ctrl+f11 twice) and checking the update and draw graph for any spikes.

**Reply by SevenChords** [reply]
Thanks for the clarification, i guess my stuttering is a screen issue rather than a game/performance issue.
